web/impact/impact/views/calendar_reminder_view.py

```diff
@@ -0,0 +1,52 @@
+from django.views import View
+from django.http import HttpResponseRedirect
+from add2cal import Add2Cal
+import datetime
+from django.http import (
+    JsonResponse,
+    HttpResponse
+)
+
+DATE_FORMAT = "%Y%m%dT%H%M%S"
+
+
+class CalendarReminderView(View):
+
+    def get(self, request, *args, **kwargs):
+        params = self.request.GET
+        start = params.get('start', datetime.datetime.now().strftime(
+            DATE_FORMAT))
+        end = params.get('end', datetime.datetime.now().strftime(DATE_FORMAT))
+        title = params.get('title', 'new reminder')
+        description = params.get('description', '')
+        location = params.get('location', 'Boston, MA')
+        link_type = params.get('link_type', 'data')
+
+        add2cal = Add2Cal(
+            start=start,
+            end=end,
+            title=title,
+            description=description,
+            location=location)
+        calendar_data = add2cal.as_dict()
+        if link_type == 'ical':
```

Comment:
This pseudo-case statement would be nicer as a dispatch dictionary. Something like:

```
dispatch_dict = {
    'ical': ical_response,
    'outlook': outlook_response,
    'google': gcal_response}

...
    def get(self, request, *args, **kwargs):
    ...
    response_function = dispatch_dict.get(link_type, default_response)
    return response_function(title)
```

Among other things, this would clean up the issue of function length here.